scriptmodules/supplementary/custombluez.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="custombluez"
+rp_module_desc="Updated version of BlueZ Bluetooth stack\n\nInstall alongside 'sixaxis' driver if you need to pair third-party (Gasia/Shanwan) DualShock 3 controllers."
+rp_module_licence="GPL2 http://www.bluez.org/faq/common/"
+rp_module_section="driver"
+
+function _version_custombluez() {
+    echo "5.48-1~rpi1"
+}
+
+function depends_custombluez() {
+    local depends=(check dh-systemd git-buildpackage icu-devtools libcap-ng-dev libdbus-1-dev libdbus-glib-1-dev libdw-dev libical-dev libicu-dev libreadline-dev libsubunit-dev libtinfo-dev)
+
+    getDepends "${depends[@]}"
+}
+
+function sources_custombluez() {
+    # git-buildpackage requires the master *and* pristine-tar branch to be available, so clone the entire repo
+    __persistent_repos=1 gitPullOrClone "$md_build/custombluez" https://github.com/psyke83/bluez-retropie.git
+}
+
+function build_custombluez() {
+    pushd custombluez
+    rm -rf "$md_build/build-area"
+    CFLAGS="${CFLAGS//-O2/-Os}" gbp buildpackage -us -uc
```

Comment:
Reverting to the lower optimization level appears to be necessary to avoid this build error:
```
/usr/bin/ld: client/main.o: relocation R_ARM_MOVW_ABS_NC against `a local symbol' can not be used when making a shared object; recompile with -fPIC
```

This happens on jessie but not stretch, as far as I can see. Running git-buildpackage from a shell manually on jessie will build the packages successfully (as ```-O2``` is not inserted via the CFLAGS environment variables).

Reply:
Will change that in the next refresh, thanks.

There's another issue that will need discussing before this PR is suitable to be included. Currently, the new driver exposes both digital and analog events for both of the trigger buttons (L2 & R2). The script currently sets a large axis fuzz on the analog axes to effectively disable them, so that only the digital inputs would be used, but disabling the analog inputs in this way will cause problems with Kodi, as they prefer to use the analog configuration (see: https://github.com/xbmc/peripheral.joystick/pull/143). Kodi has an "ignore input" feature to assist configuration of the controller: https://kodi.wiki/view/HOW-TO:PlayStation_4_controller

We might need to implement something similar for EmulationStation, otherwise it will be difficult to configure DS3 or DS4 pads correctly, as pressing L2 or R2 will register both digital and analog inputs simultaneously, causing the input configuration to assign the wrong key to the next button in the list erroneously.